### PR TITLE
Add '-web-documentation' to start a simple web server with documentation

### PR DESCRIPTION
**Pull request by 2bllw8**
Add a web documentation handler to provide documentation for easy consultation.

- `/documentation`: home page with all links
- `/documentation/handbook`: html version of the gm handbook
- `/documentation/gacha`: json document with the gacha mappings

More pages can be easily added

## Type of changes

- [ ] Bug fix
- [x] New feature 
- [ ] Enhancement
- [x] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future

### CODE REVIEW

**Comment by memetrollsXD** [comment]
Why would there be a need for a built in documentation webserver if there's already the Grasscutter wiki and other resources?

**Comment by 4Benj** [comment]
> Why would there be a need for a built in documentation webserver if there's already the Grasscutter wiki and other resources?

^^^
and why create a new server when you can use the Http/Dispatch server

**Comment by 2bllw8** [comment]
> Why would there be a need for a built in documentation webserver if there's already the Grasscutter wiki and other resources?

A web documentation unlike the wiki can give the exact information for the running version (maybe the version being used is old and the wiki has newer information that do not work on older version or vice versa).

Moreover for example if the available commands change (for example some are disabled in a specific server instance) a live version of the documentation can reflect the changes immediately

**Review by ghost — APPROVED** [review]

**Comment by mcxiaoke** [comment]
attach the DocumentationServerHandler to current http server , you can serve online docs when server running at / or /docs/, no extra command needed. 

```
httpServer.addRouter(DocumentationServerHandler.class); // just add one line in Grasscutter.java
```

**Comment by 2bllw8** [comment]
> attach the DocumentationServerHandler to current http server , you can serve online docs when server running at / or /docs/, no extra command needed.
> 
> ```
> httpServer.addRouter(DocumentationServerHandler.class); // just add one line in Grasscutter.java
> ```

Ok will update to do this, thanks :+1:

File: src/main/java/emu/grasscutter/documentation/DocumentationHandler.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package emu.grasscutter.documentation;
```
Comment (4Benj) [inline src/main/java/emu/grasscutter/documentation/DocumentationHandler.java]: The documentation package should be within ``emu.grasscutter.server.http``

**Review by 4Benj — CHANGES_REQUESTED** [review]

**Review by KingRainbow44 — CHANGES_REQUESTED** [review]
resolve Benj's request first

File: src/main/java/emu/grasscutter/documentation/DocumentationHandler.java (hunk 1 above)
Comment (2bllw8) [reply src/main/java/emu/grasscutter/documentation/DocumentationHandler.java]: Done

**Review by ghost — APPROVED** [review]

**Pull request merged**